src/microbe_stage/MicrobeStage.cs

```diff
@@ -849,6 +857,7 @@ private void HandlePlayerRespawn()
         else
         {
             // Player is not extinct, so can respawn
+            spawner.DespawnAll();
```

Comment:
This should not be required with the despawn fixes... if this is required please add a comment detailing why this is needed. I think this'll make things visually pop out of existance on camera so I think things will look better if we don't do this and instead rely on the more powerful despawn system to be able to get things done.

Reply:
I've moved this in order to keep the entities from disappearing while the player might see, but the despawn definitely is needed with this spawning system, since without it the game will hold twice as many entities as the max for a couple seconds.